backend/src/main/java/com/alex/eshop/filterSpecifications/CategorySpecification.java

```diff
@@ -0,0 +1,11 @@
+package com.alex.eshop.filterSpecifications;
+
+import com.alex.eshop.entity.Category;
+import org.springframework.data.jpa.domain.Specification;
+
+public class CategorySpecification {
```

Comment:
Usually I'm not mixing Specification with @Query annotations. You can do the same in @Query(), and it can be less code than in specification